lib/src/rewrite.rs

```diff
@@ -58,9 +64,10 @@ pub fn merge_commit_trees_without_repo(
                 .iter()
                 .map(|id| store.get_commit(id))
                 .try_collect()?;
-            let ancestor_tree = merge_commit_trees_without_repo(store, index, &ancestors)?;
+            let ancestor_tree =
+                merge_commit_trees_no_resolve_without_repo(store, index, &ancestors)?;
             let other_tree = other_commit.tree()?;
-            new_tree = new_tree.merge(&ancestor_tree, &other_tree)?;
+            new_tree = new_tree.merge_no_resolve(&ancestor_tree, &other_tree);
```

Comment:
@martinvonz Is this assumption valid?

> Since "ancestor_tree" isn't resolved by itself, this patch has subtle behavior
> change. For example, "jj diff -r9eaef582" in the "git" repository is no longer
> empty. I think the new behavior is also technically correct, but I'm not pretty
> sure.

Reply:
I think that may be due to the `A+(A-B)=A` rule resulting being applied in the ancestor before this patch (for those not familiar, that's https://github.com/martinvonz/jj/blob/bf2cb55ea7475adb36a93c8632979f956bbdedbf/lib/src/merge.rs#L92-L105 and also line 65 in that file). I've confirmed that removing that rule results in the same diff as with this patch (plus a ton of no-op "Resolved conflict" diffs). So I think in some sense the new behavior is more correct.

Reply:
Good find, btw!